java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -2032,44 +2036,58 @@ public static String uuidToLittleEndian(String uuidIn) {
      * @param key the json key of the message to decode (e.g.: sdterr, stdout)
      * @return the String decoded if it exists
      */
-    public static Optional<String> decodeStdMessage(Object message, String key) {
+    public static String decodeStdMessage(Object message, String key) {
+        JsonElement json = null;
         if (message instanceof JsonParsingError) {
-            JsonElement json = ((JsonParsingError)message).getJson();
-            if (json.isJsonObject() && json.getAsJsonObject().has(key)) {
-                if (json.getAsJsonObject().get(key).isJsonPrimitive() &&
-                    json.getAsJsonObject().get(key).getAsJsonPrimitive().isString()) {
-                    return Optional.ofNullable(
-                            json.getAsJsonObject().get(key).getAsJsonPrimitive().getAsString())
-                            .filter(s -> !s.isEmpty());
-                }
-                else if (json.getAsJsonObject().get(key).isJsonArray()) {
-                    StringBuilder msg = new StringBuilder();
-                    json.getAsJsonObject().get(key).getAsJsonArray()
-                            .forEach(elem -> msg.append(elem.getAsString()));
-                    return Optional.ofNullable(msg.toString()).filter(s -> !s.isEmpty());
-                }
+            json = ((JsonParsingError)message).getJson();
+        }
+        else if (message instanceof SaltSSHError) {
+            try {
+                // Try parsing the error message as JSON
+                json = JsonParser.GSON.fromJson("{" + ((SaltSSHError) message).getMessage() + "}", JsonElement.class);
+            }
+            catch (JsonParseException ex) {
+                LOG.debug("Unable to parse SaltSSHError message as json");
```

Comment:
maybe better a warning? Otherwise we have no idea that an error happened when looking at the logs.
Is there a way to raw dump the message? Would it make sense ?

Reply:
Indeed. I refactored the code to propagate the actual error message to the frontend even when it is not possible to parse it as json. Since now the error is correctly reported I don't think we need to have this log as a warning, but I upgraded it to info.